logstash-core/spec/logstash/webserver_spec.rb

```diff
@@ -36,29 +41,56 @@ def free_ports(servers)
     Thread.abort_on_exception = @abort
   end
 
-  let(:logger) { double("logger") }
+  let(:logger) { LogStash::Logging::Logger.new("testing") }
   let(:agent) { double("agent") }
   let(:webserver) { double("webserver") }
 
   before :each do
-    [:info, :warn, :error, :fatal, :debug].each do |level|
-      allow(logger).to receive(level)
-    end
-    [:info?, :warn?, :error?, :fatal?, :debug?].each do |level|
-      allow(logger).to receive(level)
-    end
-
     allow(webserver).to receive(:address).and_return("127.0.0.1")
     allow(agent).to receive(:webserver).and_return(webserver)
   end
 
-  context "when the port is already in use and a range is provided" do
-    subject { LogStash::WebServer.new(logger,
-                                      agent,
-                                      { :http_host => "localhost", :http_ports => port_range
-                                      })}
+  subject { LogStash::WebServer.new(logger,
+                                    agent,
+                                    { :http_host => "localhost", :http_ports => port_range })}
+
+  let(:port_range) { 10000..10010 }
+
+  context "when an exception occur in the server thread" do
+    let(:spy_output) { spy("stderr").as_null_object }
+
+    it "should not log to STDERR" do
+      backup_stderr = STDERR
```

Comment:
I did that previously and was hit by, the `variable was use in multiple example` bug :(